In [6]:
import joblib
import pandas as pd
from sklearn.metrics import cohen_kappa_score

best_rf = joblib.load("modelo_random_forest.pkl")
best_svm = joblib.load("modelo_linearsvc.pkl")
scaler = joblib.load("scaler_svm.pkl")  

df = pd.read_csv("06 - dataset_limpio.csv")

In [7]:

# Preparar datos de entrada
X = df.drop("Diabetes_binary", axis=1)
y = df["Diabetes_binary"]

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)

# Escalar para SVM
X_test_scaled = scaler.transform(X_test)

# Predicciones
y_pred_rf = best_rf.predict(X_test)
y_pred_svm = best_svm.predict(X_test_scaled)

# Análisis de concordancia
kappa_score = cohen_kappa_score(y_pred_rf, y_pred_svm)
print(f"Cohen's Kappa: {kappa_score:.4f}")

Cohen's Kappa: 0.8462


 Conclusión de la comparación entre modelos
La comparación entre los modelos Random Forest y LinearSVC ajustados muestra un alto grado de concordancia, evidenciado por un valor de Cohen's Kappa de 0.8462, lo cual se interpreta como una concordancia casi perfecta.

Este nivel de acuerdo sugiere que, aunque los dos modelos utilizan enfoques distintos (árboles de decisión vs. separación lineal), ambos están clasificando de manera muy similar los casos en el conjunto de prueba, especialmente tras haber sido optimizados para mejorar la sensibilidad hacia la clase positiva (diabetes).

En resumen, ambos modelos se comportan de forma coherente entre sí, lo cual refuerza la robustez de los resultados obtenidos y brinda confianza en su uso para la toma de decisiones en contextos sensibles como la salud pública.